dev/sql_auditing.h

```diff
@@ -0,0 +1,146 @@
+#pragma once
+
+#include <fstream>
+#include <chrono>
+#include <ctime>  // std::localtime and std::tm and std::time_t
+#include <iomanip>  // std::put_time
+#include <memory>
+#include <string>
+#include <memory>
+#include "error_code.h"
+
+enum class auditing_behavior : signed char { OFF = 0, ON = 1 };
+
+namespace sqlite_orm {
+    namespace internal {
+        class storage_base;
+    }
+}
+using sqlite_orm::internal::storage_base;
```

Comment:
why this `using` is here?

Reply:
using is used here because log_settings and log_admin are declared in the glocal namespace

Reply:
don't use global namespace in the lib code